### ISSUE TITLE
Can't use replaceWith of referencePath‘s .parentPath.parentPath from getBinding()

### ISSUE THREAD

**Issue by Ercilan** [babel/babel#12943]
## Bug Report

- [ ] I would like to work on a fix!

**Current behavior**

I want to replace a **nested callexpression** to their real code, so I cloned a new node and changed it. 

But I got an error: `ReferenceError: Container is falsy` when used `p.parentPath.parentPath.replaceWith(newNode);`.

**Input Code**

The nseted callexpression：
```js
var r = {};
r.foo(r.bar('x'));
```
But it will work when the target code is `r.foo(r.bar('x') + 'x');`

visitor:
```js
const defaultVisitor = {
    Program(path) {
        const binding = path.scope.getBinding("r");

        if (binding && binding.referenced) {
            binding.referencePaths.forEach(p => {
                // In fact, these two calls will be replaced with different complex nodes.
                const newNode = t.stringLiteral("example node");
                p.parentPath.parentPath.replaceWith(newNode);
            });
        }
    }
}
```

**Expected behavior**

```js
var r = {};
"example node";
```
Just example code.

**Environment**

```
System:
  OS: Windows 10 10.0.19042
Binaries:
  Node: 14.15.2 - C:\Program Files\nodejs\node.EXE
  npm: 6.14.9 - C:\Program Files\nodejs\npm.CMD
```
- Babel version(s): 7.13.8
- Node/npm version: Node v14.15.2
- OS:  Windows 10
- Monorepo: none
- How you are using Babel: Nodejs

**Possible Solution**

**Additional context**
This happens only when use `.parentPath.parentPath`. And it will work when target code is `r.foo(r.bar('x') + 'x')`.

After some tests, I found that the container will be assigned to null in the source code: [babel/context.ts at 233b17c4df0d42a5b0264f4ff5d4ddb9cc3effe0 · babel/babel](https://github.com/babel/babel/blob/233b17c4df0d42a5b0264f4ff5d4ddb9cc3effe0/packages/babel-traverse/src/path/context.ts#L203) line 210 (probably is useless for you😑)

---
This situation is very similar to: [Can't replace callExpression with a string literal · Issue #1764 · babel/babel](https://github.com/babel/babel/issues/1764)
Maybe：[parentPath refers to grandParent instead of parent · Issue #7596 · babel/babel](https://github.com/babel/babel/issues/7596)

As a rookie with poor english, I am very puzzled😰. Is there any other available methods to accomplish my purpose?

**Comment by Numenorean** [comment]
Similar task, same issue. Did anyone find a workground?

**Comment by liuxingbaoyu** [comment]
@Numenorean You could try converting the parent node replacement logic to child node replacement, that should avoid the problem.

**Comment by Numenorean** [comment]
> @Numenorean You could try converting the parent node replacement logic to child node replacement, that should avoid the problem.

Yeah, it clearly would be a solution, but I wonder if it somehow possible to replace "parent of parent" while iterating through the references

**Comment by liuxingbaoyu** [comment]
Currently there is no reliable way.

**Comment by Numenorean** [comment]
> Currently there is no reliable way.

Oh, I got it. Thank you. Will consider traversing directly on CallExpression

**Comment by echo094** [comment]
The `ReferenceError: Container is falsy` is thrown when handling the second reference, whose grandparent is `r.bar('x')`. In fact, the parent of `r.bar('x')`, aka `r.foo(r.bar('x'))`, is replaced by the stringLiteral "example node" after handling the first reference. Hence, its container cannot be found and the error is thrown.

In the example `r.foo(r.bar('x') + 'x')`, the parent of `r.bar('x')` is `r.bar('x') + 'x'`, and  the grandparent of `r.bar('x')` is replaced by the stringLiteral "example node". Therefore, if you perform the replacement on `r.bar('x') + 'x'`, the same error will be thrown.

To avoid the error, you should update the reference record after each record:

```javascript
const defaultVisitor = {
  Program(path) {
      let binding = path.scope.getBinding("r");

      while (binding && binding.referenced) {
        console.log(`Enter, references: ${binding.references}`)
        const newNode = t.stringLiteral("example node");
        binding.referencePaths[0].parentPath.parentPath.replaceWith(newNode);
        binding.scope.crawl()
        binding = path.scope.getBinding("r")
        console.log(`Exit, references: ${binding.references}`)
      }
  }
}
```

In this example, you will find there are no references left after the first replacement.
